webapp/rust/src/main.rs

```diff
@@ -0,0 +1,1011 @@
+        .await
+        .map_err(mysql_error_to_actix_error)?;
+    let row: Option<Chair> = tx
+        .exec_first(
+            "select * from chair where id = ? and stock > 0 for update",
+            (id,),
+        )
+        .await
+        .map_err(mysql_error_to_actix_error)?;
+    if row.is_some() {
+        tx.exec_drop("update chair set stock = stock - 1 where id = ?", (id,))
+            .await
+            .map_err(mysql_error_to_actix_error)?;
+        tx.commit().await.map_err(mysql_error_to_actix_error)?;
+        Ok(HttpResponse::Ok().finish())
+    } else {
+        Ok(HttpResponse::NotFound().finish())
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Estate {
+    id: i64,
+    thumbnail: String,
+    name: String,
+    description: String,
+    latitude: f64,
+    longitude: f64,
+    address: String,
+    rent: i64,
+    #[serde(rename = "doorHeight")]
+    door_height: i64,
+    #[serde(rename = "doorWidth")]
+    door_width: i64,
+    features: String,
+    #[serde(skip)]
+    popularity: i64,
+}
+
+impl FromRow for Estate {
+    fn from_row_opt(row: mysql_async::Row) -> Result<Self, mysql_async::FromRowError> {
+        fn convert(row: &mysql_async::Row) -> Result<Estate, ()> {
+            Ok(Estate {
+                id: row.get("id").ok_or(())?,
+                thumbnail: row.get("thumbnail").ok_or(())?,
+                name: row.get("name").ok_or(())?,
+                description: row.get("description").ok_or(())?,
+                latitude: row.get("latitude").ok_or(())?,
+                longitude: row.get("longitude").ok_or(())?,
+                address: row.get("address").ok_or(())?,
+                rent: row.get("rent").ok_or(())?,
+                door_height: row.get("door_height").ok_or(())?,
+                door_width: row.get("door_width").ok_or(())?,
+                features: row.get("features").ok_or(())?,
+                popularity: row.get("popularity").ok_or(())?,
+            })
+        }
+        convert(&row).map_err(|_| mysql_async::FromRowError(row))
+    }
+}
+
+async fn get_estate_detail(
+    db: web::Data<Pool>,
+    path: web::Path<(i64,)>,
+) -> Result<HttpResponse, AWError> {
+    let id = path.0;
+
+    let mut conn = db.get_conn().await.map_err(mysql_error_to_actix_error)?;
+    let estate: Option<Estate> = conn
+        .exec_first("select * from estate where id = ?", (id,))
+        .await
+        .map_err(mysql_error_to_actix_error)?;
+
+    if let Some(estate) = estate {
+        Ok(HttpResponse::Ok().json(estate))
+    } else {
+        Ok(HttpResponse::NotFound().finish())
+    }
+}
+
+async fn post_estate(db: web::Data<Pool>, mut payload: Multipart) -> Result<HttpResponse, AWError> {
+    let mut estates = None;
+    while let Ok(Some(field)) = payload.try_next().await {
+        let content_disposition = field.content_disposition().unwrap();
+        let filename = content_disposition.get_filename().unwrap();
+        if filename == "estates" {
```

Comment:
Rust へのベンチマークを試したところ、 `POST /api/estate` のリクエストが失敗していました。
`POST /api/chair` は成功していたので、そこから考えるとここの箇所のコードを以下のものに変更すれば成功するかなと思います。

```suggestion
        let name = content_disposition.get_name().unwrap();
        if name == "estates" {
```

Reply:
たしかに。chair で直した後に estate のほうを直し損ねてました。